packages/sit-onyx/src/components/OnyxLink/types.ts

```diff
@@ -8,6 +8,12 @@ export type OnyxLinkProps = {
    * For `_blank`, the `rel="noopener noreferrer"` will be set automatically.
    */
   target?: LinkTarget;
+  /**
+   * Whether to show the external link icon.
+   * If set to `auto`, it will be shown when the `href` leads to another website
+   * and will be hidden otherwise.
+   */
+  externalIcon?: boolean | "auto";
```

Comment:
The naming suggestions to me that an icon can be chosen to use this prop, maybe we can rename it?

```suggestion
  withExternalIcon?: boolean | "auto";
```